### ISSUE TITLE
Bug: [consistent-generic-constructors] doesn't report on `accessor` class properties

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10786]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEAq0G8BQ1oHo3QE4FMAOA9lgC44Am0wRuwxIAnitKRMQIwBc0AsmHgDyssASwB2AcwA00UQFcAtgCMcWAHzQAvDJwB3HnwAUASgDcSJhhkFi2fEVIVFsmxAAWBWSEc4mYYMBwoImZA4gAmLl4BITEpGQVlNU1tPSjjMwBfJCA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQYA2uGw4x0BZAA0a9ZCl9Z86JH1gAumoC%2BIG0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class T {
  // reported correctly
  test1: Map<string, number> = new Map();

  // not reported but should be
  accessor test2: Map<string, number> = new Map();
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-generic-constructors": [
      "error",
      "constructor"
    ],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the rule to report on the `accessor` property.

### Actual Result

The `accessor` property didn't get reported.

### Additional Info

See https://github.com/typescript-eslint/typescript-eslint/issues/10763#issuecomment-2634900340.
